specifications/android.md

```diff
@@ -82,32 +91,45 @@ There is a third action the connector SHOULD handle:
 ### org.unifiedpush.android.connector.NEW_ENDPOINT
 
 The distributor MUST send this action to the registered application to confirm the registration of an end user application, when a registered application send again an action with the intent org.unifiedpush.android.distributor.REGISTER and a valid token, or when the endpoint change with the 2 following extras:
-* token: the token supplied by the end user application during registration
-* endpoint: the endpoint
+* token (String): the token supplied by the end user application during registration
+* endpoint (String): the endpoint URL
 
 ### org.unifiedpush.android.connector.REGISTRATION_FAILED
 
-The distributor MUST send this action to the registered application if the token is already registered for another application or if the registration can not be processed (for instance when the distributor is not connected to its provider server) with the 2 following extras:
-* token: the token supplied by the end user application during registration
-* message: this extra MAY be sent to gives an error message
+The distributor MUST send this action to the registered application if:
+* the token is already registered for another application
+* the registration can not be processed (for instance when the distributor is not connected to its provider server)
+* a requested feature is not supported by the distributor
+
+The action contains the 2 following extras:
+* token (String): the token supplied by the end user application during registration
+* message (String): this extra MAY be sent to gives an error message
 
 The connector MUST change the connection token received with this action for the next registration.
 
 ### org.unifiedpush.android.connector.MESSAGE
 
 The distributor MUST send this action to the registered application to forward a push message received from the provider to the end user application.
 
-It MUST be send with the 2 following extras:
-* token: the token supplied by the end user application during registration
-* message: the push message sent by the application server. It MUST be the raw POST data received by the rewrite proxy.
+If the BYTES_MESSAGE feature was requested, it MUST send the following extras:
+* token (String): the token supplied by the end user application during registration
+* bytesMessage (ByteArray): the push message sent by the application server, as an array of bytes. It MUST be the raw POST data received by the rewrite proxy.
+
+If the BYTES_MESSAGE feature was requested, it MAY additionally send the message as a string:
+* message (String): the push message sent by the application server, as a string. This extra can't have characters with char code > 127.
```

Comment:
The char code >127 thing is really odd. Are you sure about that? That would mean that not even UTF-8 is supported, which is not true. 

Maybe this should just say that the message must be valid utf-8?

Reply:
We should replace that with a point at the beginning saying all the String extras are UTF-8 encoded

Reply:
My newly found knowledge about utf-8 tells me that only chars 127-192 are not allowed. I thought we said we'll remove this piece.

Reply:
Indeed I forgot to remove that after adding the general statement, done.

Reply:
> My newly found knowledge about utf-8 tells me that only chars 127-192 are not allowed.

Should we put that as a restriction instead of UTF-8 ?

Reply:
I don't think so. It just confuses people. I also don't know if maybe be the encoding doesn't itself contain those characters. It is just disallowed in the first byte of a code point as far as I understand. Just remove it. I think the comment at the top covers everything.

Reply:
I was talking about the comment at the top, this is already removed :) ok let's keep it like that.